vmdk_plugin/utils/fs/fs.go

```diff
@@ -185,46 +231,11 @@ func GetDevicePathByID(id string) (string, error) {
 		}
 	}
 
-	skipInotify := false
-
-	watcher, errWatcher := inotify.NewWatcher()
+	watcher, skipInotify := DevAttachWaitPrep(id, watchPath)
 
-	if errWatcher != nil {
-		log.WithFields(log.Fields{"id": id}).Error("Failed to create watcher, skip inotify ")
-		skipInotify = true
-	} else {
-		err = watcher.Watch("/dev/disk/by-id")
-		if err != nil {
-			log.WithFields(log.Fields{"id": id}).Error("Failed to watch /dev, skip inotify ")
-			skipInotify = true
-		}
-	}
 	device := makeDevicePathWithID(id)
 
-loop:
-	for {
-		select {
-		case ev := <-watcher.Event:
-			log.Debug("event: ", ev)
-			if ev.Name == device {
-				// Log when the device is discovered
-				log.WithFields(
-					log.Fields{"device": device, "event": ev},
-				).Info("Scan complete ")
-				break loop
-			}
-		case err := <-watcher.Error:
-			log.WithFields(
-				log.Fields{"device": device, "error": err},
-			).Error("Hit error during watch ")
-			break loop
-		case <-time.After(devWaitTimeout):
-			log.WithFields(
-				log.Fields{"timeout": devWaitTimeout, "device": device},
-			).Warning("Reached timeout while waiting for device ")
-			break loop
-		}
-	}
+	DevAttachWait(watcher, id, device)
```

Comment:
same question -  if skipNotify is true, then watcher will be not configured and DevAttachWait() probably should be called, right ?